src/bech32.cpp

```diff
@@ -370,11 +374,12 @@ std::string Encode(Encoding encoding, const std::string& hrp, const data& values
 }
 
 /** Decode a Bech32 or Bech32m string. */
-DecodeResult Decode(const std::string& str) {
+DecodeResult Decode(const std::string& str, CharLimit limit) {
     std::vector<int> errors;
     if (!CheckCharacters(str, errors)) return {};
     size_t pos = str.rfind('1');
-    if (str.size() > 90 || pos == str.npos || pos == 0 || pos + 7 > str.size()) {
+    if (str.size() > limit) return {};
+    if (pos == str.npos || pos == 0 || pos + 7 > str.size()) {
```

Comment:
is this 7 actually a `CHECKSUM_SIZE + 1`, or completely unrelated?
```suggestion
    if (pos == str.npos || pos == 0 || pos + CHECKSUM_SIZE >= str.size()) {
```

Reply:
Yes, this is checking that "The data part, which is at least 6 characters long" (from BIP173). Changed `pos + 7 >` to `pos + CHECKSUM_SIZE >=` throughout.